src/yadmark/benchmark/external/apfelpy_utils.py

```diff
@@ -16,6 +16,69 @@
     "total": 0,
 }
 
+PROJECTILE_PIDS = {
+    "electron": 11,
+    "positron": -11,
+    "neutrino": 12,
+    "antineutrino": -12,
+}
+
+
+def map_apfelpy_sf(init, observables, fns):
+    """Get required structure function.
+
+    Parameters
+    ----------
+    init: ap.initializers
+        Apfel++ initializers
+    observables: dict
+        observables runcard
+    fns: str
+        flavor number scheme
+
+    Returns
+    -------
+        Apfel++ structure function initalizer
+    """
+    MAP_ZM_CC = {
+        "F2m": init.InitializeF2CCMinusObjectsZM,
+        "FLm": init.InitializeFLCCMinusObjectsZM,
+        "F3m": init.InitializeF3CCMinusObjectsZM,
+        "F2p": init.InitializeF2CCPlusObjectsZM,
+        "FLp": init.InitializeFLCCPlusObjectsZM,
+        "F3p": init.InitializeF3CCPlusObjectsZM,
+    }
+
+    MAP_ZM_NC = {
+        "F2": init.InitializeF2NCObjectsZM,
+        "FL": init.InitializeFLNCObjectsZM,
+        "F3": init.InitializeF3NCObjectsZM,
+        "g1": init.Initializeg1NCObjectsZM,
+        "gL": init.InitializegLNCObjectsZM,
+        "g4": init.Initializeg4NCObjectsZM,
+    }
+
+    MAP_FFNS_NC = {
+        "F2": init.InitializeF2NCObjectsMassive,
+        "FL": init.InitializeFLNCObjectsMassive,
+    }
+    MAP_FFNS0_NC = {
+        "F2": init.InitializeF2NCObjectsMassiveZero,
+        "FL": init.InitializeFLNCObjectsMassiveZero,
+    }
+
+    if observables["prDIS"] == "CC":
+        if PROJECTILE_PIDS[observables["ProjectileDIS"]] > 0:
+            return MAP_ZM_CC
+        return ValueError(f"{observables['ProjectileDIS']} not available in Apfel++")
```

Comment:
It's not strictly an error, but I guess you wanted 
```suggestion
        raise ValueError(f"{observables['ProjectileDIS']} not available in Apfel++")
```